esp-wifi/.cargo/config.toml

```diff
@@ -5,42 +5,25 @@
 #   - `embassy-executor-thread` on Xtensa chips to take advantage of the Xtensa specific executor we have in esp-hal
 [alias]
 esp32 =   "run --features   esp32 --target xtensa-esp32-none-elf        --features esp32-hal/default,esp32-hal/embassy-time-timg0,esp32-hal/embassy-executor-thread"
-esp32s2 = "run --features esp32s2 --target xtensa-esp32s2-none-elf      --features esp32s2-hal/default,esp32s2-hal/embassy-time-timg0,esp32s2-hal/embassy-executor-thread"
+esp32s2 = "run --features esp32s2 --target xtensa-esp32s2-none-elf      --features esp32s2-hal/default,esp32s2-hal/embassy-time-timg0,esp32s2-hal/embassy-executor-thread,portable-atomic/critical-section"
```

Comment:
@MabezDev are you fine with me adding this? We enable `critical-section` in the dev dependencies, but for the simple `cargo check` it doesn't get picked up and rustc freaks out (missing compare_exchange_weak and co. because portable-atomic is missing a feature)